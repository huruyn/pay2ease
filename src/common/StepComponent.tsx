interface Steps {
  steps: number;
  currentStep: number;
  setCurrentStep: any;
  navigateSteps: any;
}

const StepComponent = ({
  steps,
  currentStep,
  setCurrentStep,
  navigateSteps,
}: Steps): JSX.Element => {
  const KeyPressHandler = () => {};
  const stepNumberStyle = {
    width: 25,
    height: 25,
    borderRadius: 50,
    border: '2px solid grey',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  };
  const selectedStepNumberStyle = {
    width: 25,
    height: 25,
    borderRadius: 50,
    border: '2px solid #C41F3E',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#C41F3E',
    color: 'white',
  };
  const allSteps = [];
  for (let i = 0; i < steps; i += 1) {
    if (i > 0 && i < steps) {
      allSteps.push(
        <div
          style={
            currentStep > i
              ? { height: 2, width: 15, backgroundColor: '#C41F3E' }
              : { height: 2, width: 15, backgroundColor: 'grey' }
          }
          key={`dash_${i}`}
        />,
      );
    }
    allSteps.push(
      // eslint-disable-next-line jsx-a11y/interactive-supports-focus
      <div
        style={currentStep >= i + 1 ? selectedStepNumberStyle : stepNumberStyle}
        role="button"
        onKeyPress={KeyPressHandler}
        onClick={() => {
          setCurrentStep(i + 1);
          if (navigateSteps) {
            navigateSteps(i);
          }
        }}
        key={`step_${i}`}
      >
        {i + 1}
      </div>,
    );
  }
  return (
    <div className="d-flex align-items-center mt-3 mb-3">
      <span style={{ fontWeight: 'bold', marginRight: 10 }}>Steps: </span>
      {allSteps}
    </div>
  );
};

export default StepComponent;
