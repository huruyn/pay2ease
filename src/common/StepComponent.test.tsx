import { render, fireEvent } from '@testing-library/react';
import StepComponent from './StepComponent';

const ReactTestRenderer = require('react-test-renderer');

describe('Step Component', () => {
  it('matches the snapshot', () => {
    const setCurrentStep = jest.fn();
    const steps = 3;
    const tree = ReactTestRenderer.create(
      <StepComponent
        steps={steps}
        currentStep={1}
        setCurrentStep={setCurrentStep}
        navigateSteps={jest.fn()}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Click Step button on Step Component', () => {
    const setCurrentStep = jest.fn();
    const steps = 3;
    const { container } = render(
      <StepComponent
        steps={steps}
        currentStep={1}
        setCurrentStep={setCurrentStep}
        navigateSteps={jest.fn()}
      />,
    );
    const mEvent = { preventDefault: jest.fn() };
    const btnElements = container.querySelectorAll('div[role="button"]');
    expect(btnElements).toHaveLength(steps);
    fireEvent.click(btnElements[0], mEvent);
    fireEvent.click(btnElements[1], mEvent);
    fireEvent.click(btnElements[2], mEvent);
    expect(setCurrentStep).toBeCalledTimes(steps);
  });
});
