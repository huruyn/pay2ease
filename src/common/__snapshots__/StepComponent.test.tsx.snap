// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step Component matches the snapshot 1`] = `
<div
  className="d-flex align-items-center mt-3 mb-3"
>
  <span
    style={
      Object {
        "fontWeight": "bold",
        "marginRight": 10,
      }
    }
  >
    Steps: 
  </span>
  <div
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#C41F3E",
        "border": "2px solid #C41F3E",
        "borderRadius": 50,
        "color": "white",
        "display": "flex",
        "height": 25,
        "justifyContent": "center",
        "width": 25,
      }
    }
  >
    1
  </div>
  <div
    style={
      Object {
        "backgroundColor": "grey",
        "height": 2,
        "width": 15,
      }
    }
  />
  <div
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "alignItems": "center",
        "border": "2px solid grey",
        "borderRadius": 50,
        "display": "flex",
        "height": 25,
        "justifyContent": "center",
        "width": 25,
      }
    }
  >
    2
  </div>
  <div
    style={
      Object {
        "backgroundColor": "grey",
        "height": 2,
        "width": 15,
      }
    }
  />
  <div
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "alignItems": "center",
        "border": "2px solid grey",
        "borderRadius": 50,
        "display": "flex",
        "height": 25,
        "justifyContent": "center",
        "width": 25,
      }
    }
  >
    3
  </div>
</div>
`;
